### PR TITLE
Custom attributes support

### PR DESCRIPTION
**Pull request by hashedone**

### CODE REVIEW

**Comment by hashedone** [comment]
This is a thing I would really need for my project (https://github.com/CosmWasm/sylvia) so I forked your crate and implemented it. If you want to include it in your official crate I would be more than happy. If you have any suggestions - feel free to tell me.

File: src/lib.rs [hunk 1]
```diff
@@ -275,6 +279,90 @@ where
...
+///
+/// use quote::quote;
+/// use syn::parse_macro_input;
+///
+/// #[proc_macro_attribute]
+/// fn hello(attr: proc_macro::TokenStream, item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+///     hello_internal(attr.into(), item.into()).into()
+/// }
+///
+/// fn hello_intenrnal(attr: proc_macro2::TokenStream, item: proc_macro2::TokenStream) -> proc_macro2::TokenStream {
```
Comment (tonyfinn) [inline src/lib.rs:322]: Should be `hello_internal` and not `hello_intenrnal`?

File: src/lib.rs [hunk 2]
```diff
@@ -275,6 +279,90 @@ where
...
 
+/// Parses the given Rust source file, finding attributes macro expansions using `macro_path`.
+/// Each time it finds one, it calls `derive_fn`, passing it a `syn::DeriveInput`.
+///
+/// Note that this parser only handles Rust's syntax, so it cannot resolve paths to see if they
+/// are equivalent to the given one. The paths used to reference the macro must be exactly equal
+/// to the one given in order to be expanded by this function. For example, if `macro_path` is
+/// `"foo"` and the file provided calls the macro using `#[bar::foo]`, this function will not know
+/// to expand it, and the macro's code coverage will be underestimated. Also it is important, that
+/// this function would expand every matching attribute, so it is important to desing your macros
```
Comment (tonyfinn) [inline src/lib.rs:290]: Nitpick: `desing` -> `design`

**Review by tonyfinn — APPROVED** [review]
Looks good to me, two small typos to fix and then I'll merge it

**Comment by tonyfinn** [comment]
This is now released as v0.6.0

**Pull request merged**